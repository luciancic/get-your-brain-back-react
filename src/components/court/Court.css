.court {
    height: 100vh;
    display: grid;
    grid-template-rows: 15% 60% 15% 10%;
    align-items: center;
    justify-items: center;
}

@media (max-height: 500px) and (min-aspect-ratio: 16/10) {
    .court {
        grid-template-rows: 20% 75%;
        grid-template-columns: 70% 25%;
    }

    .court-buttons {
        grid-column: 2;
        grid-row: 1/3;
    }
}