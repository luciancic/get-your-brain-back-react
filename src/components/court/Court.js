// Component shared across Game and Tutorial
import React, { Component } from 'react';
import IndicatorBar from './IndicatorBar';
import Letters from './Letters';
import MatchButton from './MatchButton';
import PositionBoard from './PositionBoard';
import escapeable from '../escapeable';

import './Court.css';

class Court extends Component {
    render() {
        const { 
            audioPlayed, 
            letters, 
            lettersButtonColor,
            maxRounds, 
            n,
            positionHandler,
            lettersHandler,
            playAudio, 
            positions, 
            positionsButtonColor, 
            remainingRounds, 
            roundActive, 
        } = this.props;

        return <div className="court">
            <IndicatorBar maxRounds={maxRounds} remainingRounds={remainingRounds} n={n} />
            <PositionBoard active={roundActive} currentPosition={positions[0]}/>
            <div className="court-buttons">
                {/* TODO: extract match function to make it usable in tutorial */}
                <MatchButton color={positionsButtonColor} name="Match Position" onClick={positionHandler}/>
                <MatchButton color={lettersButtonColor} name="Match Letter" onClick={lettersHandler}/>
                {/* <button className={`btn ${positionsButtonColor} lighten-4 blue-text text-darken-4`}>Match Position</button>
                <button className={`btn ${lettersButtonColor} lighten-4 blue-text text-darken-4`}>Match Letter</button> */}
            </div>
            <Letters roundActive={roundActive} currentLetter={letters[0]} audioPlayed={audioPlayed} playAudio={playAudio}/>
        </div>
    }
}

export default escapeable(Court);