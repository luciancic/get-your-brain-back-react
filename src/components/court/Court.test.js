import React from 'react';
import { shallow } from 'enzyme';
import { Court } from './Court';

it('renders Court', () => {
  expect( shallow( <Court positions={[]} letters={[]}/> )).toMatchSnapshot();
});
