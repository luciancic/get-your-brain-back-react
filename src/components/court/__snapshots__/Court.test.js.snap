// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Court 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Court
    letters={Array []}
    positions={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <_default />,
        <_default />,
        <div
          className="court-buttons"
        >
          <MatchButton
            name="Match Position"
          />
          <MatchButton
            name="Match Letter"
          />
        </div>,
        <_default />,
      ],
      "className": "court",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "maxRounds": undefined,
          "n": undefined,
          "remainingRounds": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": undefined,
          "currentPosition": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <MatchButton
              name="Match Position"
            />,
            <MatchButton
              name="Match Letter"
            />,
          ],
          "className": "court-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": undefined,
              "name": "Match Position",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": undefined,
              "name": "Match Letter",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "audioPlayed": undefined,
          "currentLetter": undefined,
          "playAudio": undefined,
          "roundActive": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <_default />,
          <_default />,
          <div
            className="court-buttons"
          >
            <MatchButton
              name="Match Position"
            />
            <MatchButton
              name="Match Letter"
            />
          </div>,
          <_default />,
        ],
        "className": "court",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "maxRounds": undefined,
            "n": undefined,
            "remainingRounds": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": undefined,
            "currentPosition": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <MatchButton
                name="Match Position"
              />,
              <MatchButton
                name="Match Letter"
              />,
            ],
            "className": "court-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": undefined,
                "name": "Match Position",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": undefined,
                "name": "Match Letter",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "audioPlayed": undefined,
            "currentLetter": undefined,
            "playAudio": undefined,
            "roundActive": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
